### PR TITLE
chore: rdev build image only

### PR DESCRIPTION
**Pull request by atolopko-czi**
https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell/255

### Reviewers
**Functional:** 
@ebezzi 

**Readability:** 
@metakuni 

---

## Changes
Pushing a branch or tag with an `rdev` prefix or suffix will now build & push (to ECR) the Docker images, but will NOT deploy to rdev. The deploy step is now performed exclusively with the `single-cell-infra` repo's `happy` command, which creates/updates an rdev stack using a specific docker image tag that must exist on both explorer and portal images.

Documented in [Google Doc](https://github.com/chanzuckerberg/single-cell-data-portal/pull/1653/commits/c3887b13ec20ecd6651a144f64a3cd3dd91bffbc). Replaces REMOTE_DEV.md, now removed.

### CODE REVIEW

File: Dockerfile.processing_image [hunk 1]
```diff
@@ -11,17 +11,17 @@ ENV PIP_NO_CACHE_DIR=1
...
- && cd single-cell-explorer \
- && make build-for-server-dev \
- && cd .. \
- && rm -rf single-cell-explorer client build/client /usr/bin/npm
+# RUN git clone https://github.com/chanzuckerberg/single-cell-explorer.git \
+#  && echo '#!/bin/sh\nmkdir -p build/static && touch build/static/.gitignore build/index.html build/csp-hashes.json' > /usr/bin/npm && chmod 755 /usr/bin/npm \
+#  && cd single-cell-explorer \
+#  && make build-for-server-dev \
+#  && cd .. \
+#  && rm -rf single-cell-explorer client build/client /usr/bin/npm
```
Comment (atolopko-czi) [inline Dockerfile.processing_image:19]: prove to me we need this...

File: .github/workflows/push-rdev.yml [hunk 2]
```diff
@@ -2,8 +2,13 @@ name: Push Remote Dev
 
 on:
   push:
+    tags:
+      - '*rdev'
+      - 'rdev*'
     branches:
       - 'rdev-*'
+      - '**/rdev-*'
```
Comment (metakuni) [inline .github/workflows/push-rdev.yml:10]: I was wondering about more flexible rdev branch names. Thank you. 🙇

File: Dockerfile.processing_image (hunk 1 above)
Comment (metakuni) [reply Dockerfile.processing_image:19]: 99.99% sure we don't need it. We'll find out for sure later today. 😄

**Review by metakuni — APPROVED** [review]

File: .github/workflows/push-rdev.yml (hunk 2 above)
Comment (metakuni) [reply .github/workflows/push-rdev.yml:10]: Ohhh, just noticed the tags, even betterer!

**Review by metakuni — APPROVED** [review]

File: Dockerfile.processing_image (hunk 1 above)
Comment (atolopko-czi) [reply Dockerfile.processing_image:19]: @metakuni Anything break?

**Review by metakuni — APPROVED** [review]

File: scripts/happy [hunk 3]
```diff
@@ -1164,6 +1164,10 @@ def push(ctx, images, tag, extra_tag, compose_env):
     # images to the right places with some re-tagging.
     first_registry = get_ecr_repo(config)
     builder_config["ecr_repo"] = first_registry  # TODO this is a leak in our abstraction, need to fix.
+    # HACK: Sets happy env name to the first extra tag; we need to know the happy env name at docker image build time, unfortunately, since it's not parameterized in the Docker image env
+    rdev_stack_name = extra_tag[0]
```
Comment (tihuan) [inline scripts/happy:1168]: @atolopko-czi it seems like this breaks running `./scripts/happy create RDEV_NAME` in Data Portal repo directly for some reason 😄 I'm currently just hacking it by hard coding `rdev_stack_name = "thuang-wmg-2"`, so I can push to rdev. So not a blocker! PTAL tomorrow thank you 🙏

Comment (atolopko-czi) [reply scripts/happy:1168]: Thanks for the bug report!  [Will investigate](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/1682).

**Comment by tihuan** [comment]
Thanks so much, Andrew 🎉 !

**Pull request merged**